lib/bson/binary.js

```diff
@@ -28,6 +28,14 @@ if (typeof global !== 'undefined') {
 function Binary(buffer, subType) {
   if (!(this instanceof Binary)) return new Binary(buffer, subType);
 
+  if (
+    buffer != null &&
+    !(typeof buffer === 'string') &&
```

Comment:
1. We should use `Buffer.isBuffer` instead of `instanceof Buffer`
2. This appears to only check if the buffer is non-null and either string or buffer. Can we add checks for `Uint8Array` and `Array`?